agent.py

```diff
@@ -188,6 +188,7 @@ async def entrypoint(self, ctx: JobContext):
 
         logger.info(f"<<<--•••-->>> participant attributes: {participant.attributes}")
 
+        await wait_for_participant_attribute(self, participant, "role")
```

Comment:
There is a great chance that the participant.attributes.get("role") returns None at this moment. If one looks at the sequence of the things happening on the client side, you will see that there are many thing that would delay. So this tries to resolve a race condition issue.